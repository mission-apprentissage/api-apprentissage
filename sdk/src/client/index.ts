export * from "./generateUri/generateUri.js";

export * from "./parser/response.parser.js";

export * from "./apiError.js";

export * from "./client.js";
