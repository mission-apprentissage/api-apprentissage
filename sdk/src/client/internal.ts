export * from "./internal/generateUri/generateUri.js";

export * from "./internal/parser/response.parser.js";

export * from "./internal/apiError.js";
