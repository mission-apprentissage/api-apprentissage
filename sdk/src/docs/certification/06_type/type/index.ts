import { DocField } from "../../../types.js";
import description from "./description.md.js";
import information from "./information.md.js";

export const typeField: DocField = {
  name: "type",
  description,
  information,
  sample: null,
  tags: [
    ".nature.cfd.code",
    ".nature.cfd.libelle",
    ".gestionnaire_diplome",
    ".certificateurs_rncp[].siret",
    ".certificateurs_rncp[].nom",
    ".enregistrement_rncp",
    ".voie_acces.rncp",
  ],
};
