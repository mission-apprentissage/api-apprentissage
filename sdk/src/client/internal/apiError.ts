import type { IResErrorJson } from "../../models/errors/errors.model.js";
import type { PathParam, QueryString, WithQueryStringAndPathParam } from "./generateUri/generateUri.js";

export interface ApiErrorContext {
  path: string;
  params: PathParam;
  querystring: QueryString;
  requestHeaders: Record<string, string | string[]>;
  statusCode: number;
  message: string;
  name: string;
  responseHeaders: Record<string, string | string[]>;
  errorData: unknown;
}

export class ApiError extends Error {
  context: ApiErrorContext;

  constructor(context: ApiErrorContext) {
    super();
    this.context = context;
    this.name = context.name ?? "ApiError";
    this.message = context.message ?? `code ${context.statusCode}`;
  }

  toJSON(): ApiErrorContext {
    return this.context;
  }

  static async build(
    path: string,
    requestHeaders: Readonly<Headers>,
    options: WithQueryStringAndPathParam,
    res: Response
  ): Promise<ApiError> {
    let message = res.status === 0 ? "Network Error" : res.statusText;
    let name = "Api Error";
    let errorData: IResErrorJson["data"] | null = null;

    if (res.status > 0) {
      try {
        if (res.headers.get("Content-Type")?.startsWith("application/json")) {
          const data = await res.json();
          if (typeof data === "object" && data !== null) {
            name = "name" in data && typeof data.name === "string" ? data.name : "Api Error";
            message = "message" in data && typeof data.message === "string" ? data.message : `code ${res.status}`;
            errorData = "data" in data ? data.data : null;
          }
        }
      } catch (_error) {
        // ignore
      }
    }

    return new ApiError({
      path,
      params: "params" in options && options.params ? options.params : {},
      querystring: "querystring" in options && options.querystring ? options.querystring : {},
      requestHeaders: Object.fromEntries(requestHeaders.entries()),
      statusCode: res.status,
      message,
      name,
      responseHeaders: Object.fromEntries(res.headers.entries()),
      errorData,
    });
  }
}
