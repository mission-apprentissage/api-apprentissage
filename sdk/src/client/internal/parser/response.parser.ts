import { z } from "zod/v4-mini";
import { $ZodError, treeifyError } from "zod/v4/core";
import type { $ZodType } from "zod/v4/core";

export class ApiParseError extends Error {
  constructor(zodError: $ZodError) {
    super(
      "api-alternance-sdk: Error while parsing API response, this is either a bug in the SDK or a change in the API response. Please report this issue.\n" +
        JSON.stringify(treeifyError(zodError))
    );
    this.name = "ApiParseError";
    this.cause = zodError;
  }
}

export function parseApiResponse<T extends $ZodType>(data: unknown, schema: T): z.output<T> {
  try {
    return z.parse(schema, data);
  } catch (error) {
    if (error instanceof $ZodError) {
      throw new ApiParseError(error);
    }
    throw error;
  }
}
