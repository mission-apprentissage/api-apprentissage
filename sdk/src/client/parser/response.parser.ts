import type { z, ZodType } from "zod";
import { ZodError } from "zod";

export class ApiParseError extends Error {
  constructor(zodError: ZodError) {
    super(
      "api-alternance-sdk: Error while parsing API response, this is either a bug in the SDK or a change in the API response. Please report this issue.\n\tZodErrors: " +
        zodError.toString().split("\n").join("\n\t")
    );
    this.name = "ApiParseError";
    this.cause = zodError;
  }
}

export function parseApiResponse<T extends ZodType>(data: unknown, schema: T): z.output<T> {
  try {
    return schema.parse(data);
  } catch (error) {
    if (error instanceof ZodError) {
      throw new ApiParseError(error);
    }
    throw error;
  }
}
