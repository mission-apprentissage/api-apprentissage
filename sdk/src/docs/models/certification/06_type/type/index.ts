import type { DocBusinessField } from "../../../../types.js";
import description from "./description.md.js";

export default <DocBusinessField>{
  type: "business",
  name: "type",
  description,
  information: null,
  sample: null,
  notes: null,
  tip: null,
  tags: [
    ".nature.cfd.code",
    ".nature.cfd.libelle",
    ".gestionnaire_diplome",
    ".certificateurs_rncp[].siret",
    ".certificateurs_rncp[].nom",
    ".enregistrement_rncp",
    ".voie_acces.rncp",
  ],
};
