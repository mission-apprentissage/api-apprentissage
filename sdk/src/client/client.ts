import type { EmptyObject } from "type-fest";
import type { z, ZodType } from "zod";

import type { IApiRouteSchema, IApiRouteSchemaWrite } from "../routes/common.routes.js";
import type {
  IApiDeleteRoutes,
  IApiGetRoutes,
  IApiPostRoutes,
  IApiPutRoutes,
  IApiRequest,
  IApiResponse,
} from "../routes/index.js";
import { ApiError } from "./apiError.js";
import type { CertificationModule } from "./certification/certification.module.js";
import { buildCertificationModule } from "./certification/certification.module.js";
import type { WithQueryStringAndPathParam } from "./generateUri/generateUri.js";
import { generateUri } from "./generateUri/generateUri.js";
import type { JobModule } from "./job/job.module.js";
import { buildJobModule } from "./job/job.module.js";
import type { OrganismeModule } from "./organisme/organisme.module.js";
import { buildOrganismeModule } from "./organisme/organisme.module.js";

type OptionsGet = {
  [Prop in keyof Pick<IApiRouteSchema, "params" | "querystring" | "headers">]: IApiRouteSchema[Prop] extends ZodType
    ? z.input<IApiRouteSchema[Prop]>
    : never;
};

type OptionsWrite = {
  [Prop in keyof Pick<
    IApiRouteSchemaWrite,
    "params" | "querystring" | "headers" | "body"
  >]: IApiRouteSchemaWrite[Prop] extends ZodType ? z.input<IApiRouteSchemaWrite[Prop]> : never;
};

type IRequestOptions = OptionsGet | OptionsWrite | EmptyObject;

type FetchOptions = Omit<RequestInit, "body" | "method" | "headers">;

const removeAtEnd = (url: string, removed: string): string =>
  url.endsWith(removed) ? url.slice(0, -removed.length) : url;

function throwError(message: string): never {
  throw new Error(message);
}

export type ApiClientConfig = {
  endpoint?: string;
  key: string;
};

export class ApiClient {
  endpoint: string;
  key: string;

  certification: CertificationModule;
  job: JobModule;
  organisme: OrganismeModule;

  constructor(config: ApiClientConfig) {
    this.endpoint = removeAtEnd(config.endpoint ?? "https://api.apprentissage.beta.gouv.fr/api", "/");
    this.key = config.key ?? throwError("api-alternance-sdk: api key is required");

    this.certification = buildCertificationModule(this);
    this.job = buildJobModule(this);
    this.organisme = buildOrganismeModule(this);
  }

  private buildRequestInit(
    method: RequestInit["method"],
    options: IRequestOptions,
    rawOptions?: FetchOptions
  ): RequestInit {
    const headers = new Headers();
    headers.set("authorization", `Bearer ${this.key}`);

    if ("headers" in options && options.headers) {
      const h = options.headers;
      Object.keys(h).forEach((name) => {
        headers.set(name, h[name]);
      });
    }

    let body: RequestInit["body"] | undefined = undefined;
    if ("body" in options && method !== "GET") {
      // @ts-expect-error
      if (options.body instanceof FormData) {
        body = options.body;
      } else {
        body = JSON.stringify(options.body);
        headers.set("content-Type", "application/json");
      }
    }

    const requestInit: RequestInit = {
      body,
      method,
      headers,
    };

    if (rawOptions) {
      Object.assign(requestInit, rawOptions);
    }

    return requestInit;
  }

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  private async exec(path: string, requestInit: RequestInit, options: WithQueryStringAndPathParam): Promise<any> {
    const reqHeaders = Array.from((requestInit.headers as Headers)?.entries() ?? []);
    const res = await fetch(this.endpoint + generateUri(path, options), requestInit);

    if (!res.ok) {
      throw await ApiError.build(path, new Headers(reqHeaders), options, res);
    }

    if (res.status === 204) {
      return;
    }

    return res.json();
  }

  async get<P extends keyof IApiGetRoutes, S extends IApiGetRoutes[P] = IApiGetRoutes[P]>(
    path: P,
    options: IApiRequest<S>,
    rawOptions?: FetchOptions
  ): Promise<IApiResponse<S>> {
    const requestInit = this.buildRequestInit("GET", options, rawOptions);
    return this.exec(path, requestInit, options);
  }

  async post<P extends keyof IApiPostRoutes, S extends IApiPostRoutes[P] = IApiPostRoutes[P]>(
    path: P,
    options: IApiRequest<S>,
    rawOptions?: FetchOptions
  ): Promise<IApiResponse<S>> {
    const requestInit = this.buildRequestInit("POST", options, rawOptions);
    return this.exec(path, requestInit, options);
  }

  async put<P extends keyof IApiPutRoutes, S extends IApiPutRoutes[P] = IApiPutRoutes[P]>(
    path: P,
    options: IApiRequest<S>,
    rawOptions?: FetchOptions
  ): Promise<IApiResponse<S>> {
    const requestInit = this.buildRequestInit("PUT", options, rawOptions);
    return this.exec(path, requestInit, options);
  }

  async delete<P extends keyof IApiDeleteRoutes, S extends IApiDeleteRoutes[P] = IApiDeleteRoutes[P]>(
    path: P,
    options: IApiRequest<S>,
    rawOptions?: FetchOptions
  ): Promise<IApiResponse<S>> {
    const requestInit = this.buildRequestInit("DELETE", options, rawOptions);
    return this.exec(path, requestInit, options);
  }
}
