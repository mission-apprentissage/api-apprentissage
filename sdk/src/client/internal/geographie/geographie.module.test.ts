import nock, { cleanAll, disableNetConnect, enableNetConnect } from "nock";
import { beforeEach, describe, expect, expectTypeOf, it } from "vitest";

import type { ICommune, IDepartement, IMissionLocale } from "../../../models/index.js";
import { ApiClient } from "../../client.js";
import { ApiError } from "../apiError.js";
import { ApiParseError } from "../parser/response.parser.js";

beforeEach(() => {
  disableNetConnect();

  return () => {
    cleanAll();
    enableNetConnect();
  };
});

describe("rechercheCommune", () => {
  const code = "59330";
  const response: ICommune[] = [
    {
      nom: "Adon",
      code: {
        insee: "45001",
        postaux: ["45230"],
      },
      localisation: {
        centre: {
          type: "Point",
          coordinates: [2.7925, 47.7587],
        },
        bbox: {
          type: "Polygon",
          coordinates: [
            [
              [2.749006, 47.727449],
              [2.835944, 47.727449],
              [2.835944, 47.790032],
              [2.749006, 47.790032],
              [2.749006, 47.727449],
            ],
          ],
        },
      },
      departement: {
        nom: "Loiret",
        codeInsee: "45",
      },
      region: {
        nom: "Centre-Val de Loire",
        codeInsee: "24",
      },
      academie: {
        nom: "Orléans-Tours",
        id: "A18",
        code: "18",
      },
      mission_locale: {
        id: 346,
        nom: "DE MONTARGIS - GIEN - AIJAM",
        siret: "43410235600012",
        localisation: {
          geopoint: {
            type: "Point",
            coordinates: [2.7459767, 48.007378],
          },
          adresse: "Centre Nelson Mandela",
          cp: "45200",
          ville: "MONTARGIS",
        },
        contact: {
          email: "contact@aijam.com",
          telephone: "02 38 98 73 60",
          siteWeb: "https://aijam.com",
        },
      },
      arrondissements: [
        { code: "75101", nom: "Paris 1er Arrondissement" },
        { code: "75120", nom: "Paris 20e Arrondissement" },
      ],
      anciennes: [
        { codeInsee: "77166", nom: "Écuelles" },
        { codeInsee: "77170", nom: "Épisy" },
      ],
    },
  ];

  it("should call the API with the correct querystring", async () => {
    const scope = nock("https://api.apprentissage.beta.gouv.fr/api", {
      reqheaders: { authorization: "Bearer api-key" },
    })
      .get("/geographie/v1/commune/search")
      .query({ code })
      .reply(200, response);

    const apiClient = new ApiClient({ key: "api-key" });

    const data = await apiClient.geographie.rechercheCommune({ code });

    expectTypeOf(data).toEqualTypeOf<ICommune[]>();

    expect(scope.isDone()).toBe(true);
    expect(data).toEqual(response);
  });

  it("should throw an ApiError when server error", async () => {
    const scope = nock("https://api.apprentissage.beta.gouv.fr/api", {
      reqheaders: { authorization: "Bearer api-key" },
    })
      .get("/geographie/v1/commune/search")
      .query({ code })
      .reply(401, {
        statusCode: 401,
        name: "Unauthorized",
        message: "Vous devez fournir une clé d'API valide pour accéder à cette ressource",
      });

    const apiClient = new ApiClient({ key: "api-key" });
    const err = await apiClient.geographie
      .rechercheCommune({ code })
      .then(() => {
        expect.unreachable("should throw an error");
      })
      .catch((error: ApiError) => {
        return error;
      });

    expect(err).toBeInstanceOf(ApiError);
    expect(err.name).toBe("Unauthorized");

    expect(scope.isDone()).toBe(true);
  });

  it("should throw if the response does not match the schema", async () => {
    const scope = nock("https://api.apprentissage.beta.gouv.fr/api", {
      reqheaders: { authorization: "Bearer api-key" },
    })
      .get("/geographie/v1/commune/search")
      .query({ code })
      .reply(200, { breaking: "schema" });

    const apiClient = new ApiClient({ key: "api-key" });
    const err = await apiClient.geographie
      .rechercheCommune({ code })
      .then(() => {
        expect.unreachable("should throw an error");
      })
      .catch((error: ApiError) => {
        return error;
      });

    expect(err).toBeInstanceOf(ApiParseError);
    expect(err.name).toBe("ApiParseError");
    expect(err.message).toMatchSnapshot();

    expect(scope.isDone()).toBe(true);
  });

  it("should accepts future schema ehancements", async () => {
    const scope = nock("https://api.apprentissage.beta.gouv.fr/api", {
      reqheaders: { authorization: "Bearer api-key" },
    })
      .get("/geographie/v1/commune/search")
      .query({ code })
      .reply(
        200,
        response.map((c) => ({
          ...c,
          new_field: "new_field",
        }))
      );

    const apiClient = new ApiClient({ key: "api-key" });

    const data = await apiClient.geographie.rechercheCommune({ code });

    expectTypeOf(data).toEqualTypeOf<ICommune[]>();

    expect(scope.isDone()).toBe(true);
    expect(data).toEqual(response);
  });
});

describe("listDepartements", () => {
  const response = [
    {
      codeInsee: "18",
      nom: "Cher",
      region: {
        codeInsee: "24",
        nom: "Centre-Val de Loire",
      },
      academie: {
        code: "18",
        id: "A18",
        nom: "Orléans-Tours",
      },
    },
    {
      codeInsee: "41",
      nom: "Loir-et-Cher",
      region: {
        codeInsee: "24",
        nom: "Centre-Val de Loire",
      },
      academie: {
        code: "18",
        id: "A18",
        nom: "Orléans-Tours",
      },
    },
    {
      codeInsee: "45",
      nom: "Loiret",
      region: {
        codeInsee: "24",
        nom: "Centre-Val de Loire",
      },
      academie: {
        code: "18",
        id: "A18",
        nom: "Orléans-Tours",
      },
    },
    {
      codeInsee: "75",
      nom: "Paris",
      region: {
        codeInsee: "11",
        nom: "Île-de-France",
      },
      academie: {
        code: "01",
        id: "A01",
        nom: "Paris",
      },
    },
    {
      codeInsee: "77",
      nom: "Seine-et-Marne",
      region: {
        codeInsee: "11",
        nom: "Île-de-France",
      },
      academie: {
        code: "24",
        id: "A24",
        nom: "Créteil",
      },
    },
    {
      codeInsee: "975",
      nom: "Saint-Pierre-et-Miquelon",
      region: {
        codeInsee: "975",
        nom: "Saint-Pierre-et-Miquelon",
      },
      academie: {
        code: "44",
        id: "A44",
        nom: "Saint-Pierre-et-Miquelon",
      },
    },
  ];

  it("should call the API", async () => {
    const scope = nock("https://api.apprentissage.beta.gouv.fr/api", {
      reqheaders: { authorization: "Bearer api-key" },
    })
      .get("/geographie/v1/departement")
      .reply(200, response);

    const apiClient = new ApiClient({ key: "api-key" });

    const data = await apiClient.geographie.listDepartements();

    expectTypeOf(data).toEqualTypeOf<IDepartement[]>();

    expect(scope.isDone()).toBe(true);
    expect(data).toEqual(response);
  });

  it("should throw an ApiError when server error", async () => {
    const scope = nock("https://api.apprentissage.beta.gouv.fr/api", {
      reqheaders: { authorization: "Bearer api-key" },
    })
      .get("/geographie/v1/departement")
      .reply(401, {
        statusCode: 401,
        name: "Unauthorized",
        message: "Vous devez fournir une clé d'API valide pour accéder à cette ressource",
      });

    const apiClient = new ApiClient({ key: "api-key" });
    const err = await apiClient.geographie
      .listDepartements()
      .then(() => {
        expect.unreachable("should throw an error");
      })
      .catch((error: ApiError) => {
        return error;
      });

    expect(err).toBeInstanceOf(ApiError);
    expect(err.name).toBe("Unauthorized");

    expect(scope.isDone()).toBe(true);
  });

  it("should throw if the response does not match the schema", async () => {
    const scope = nock("https://api.apprentissage.beta.gouv.fr/api", {
      reqheaders: { authorization: "Bearer api-key" },
    })
      .get("/geographie/v1/departement")
      .reply(200, { breaking: "schema" });

    const apiClient = new ApiClient({ key: "api-key" });
    const err = await apiClient.geographie
      .listDepartements()
      .then(() => {
        expect.unreachable("should throw an error");
      })
      .catch((error: ApiError) => {
        return error;
      });

    expect(err).toBeInstanceOf(ApiParseError);
    expect(err.name).toBe("ApiParseError");
    expect(err.message).toMatchSnapshot();

    expect(scope.isDone()).toBe(true);
  });

  it("should accepts future schema ehancements", async () => {
    const scope = nock("https://api.apprentissage.beta.gouv.fr/api", {
      reqheaders: { authorization: "Bearer api-key" },
    })
      .get("/geographie/v1/departement")
      .reply(
        200,
        response.map((c) => ({
          ...c,
          new_field: "new_field",
        }))
      );

    const apiClient = new ApiClient({ key: "api-key" });

    const data = await apiClient.geographie.listDepartements();

    expectTypeOf(data).toEqualTypeOf<IDepartement[]>();

    expect(scope.isDone()).toBe(true);
    expect(data).toEqual(response);
  });
});

describe("listMissionLocale", () => {
  const response = [
    {
      id: 1,
      nom: "DE LA PICARDIE MARITIME",
      siret: "43012526000017",
      localisation: {
        geopoint: {
          coordinates: [1.8396337, 50.099872],
          type: "Point",
        },
        adresse: "82, rue Saint-Gilles",
        cp: "80100",
        ville: "ABBEVILLE",
      },
      contact: {
        email: "mlpm@mail.fr",
        telephone: "03 22 20 14 14",
        siteWeb: "https://www.mlpm.org",
      },
    },
    {
      id: 2,
      nom: "DE L'AGENAIS, DE L'ALBRET ET DU CONFLUENT",
      siret: "38988650800040",
      localisation: {
        geopoint: {
          coordinates: [0.6231217, 44.2071179],
          type: "Point",
        },
        adresse: "70 Bld Silvain Dumon",
        cp: "47000",
        ville: "AGEN",
      },
      contact: {
        email: "accueil@mail.fr",
        telephone: "05 53 47 23 32",
        siteWeb: "https://missionlocale-agen.org/",
      },
    },
  ];

  it("should call the API", async () => {
    const scope = nock("https://api.apprentissage.beta.gouv.fr/api", {
      reqheaders: { authorization: "Bearer api-key" },
    })
      .get("/geographie/v1/mission-locale")
      .reply(200, response);

    const apiClient = new ApiClient({ key: "api-key" });

    const data = await apiClient.geographie.listMissionLocales();

    expectTypeOf(data).toEqualTypeOf<IMissionLocale[]>();

    expect(scope.isDone()).toBe(true);
    expect(data).toEqual(response);
  });

  it("should throw an ApiError when server error", async () => {
    const scope = nock("https://api.apprentissage.beta.gouv.fr/api", {
      reqheaders: { authorization: "Bearer api-key" },
    })
      .get("/geographie/v1/mission-locale")
      .reply(401, {
        statusCode: 401,
        name: "Unauthorized",
        message: "Vous devez fournir une clé d'API valide pour accéder à cette ressource",
      });

    const apiClient = new ApiClient({ key: "api-key" });
    const err = await apiClient.geographie
      .listMissionLocales()
      .then(() => {
        expect.unreachable("should throw an error");
      })
      .catch((error: ApiError) => {
        return error;
      });

    expect(err).toBeInstanceOf(ApiError);
    expect(err.name).toBe("Unauthorized");

    expect(scope.isDone()).toBe(true);
  });

  it("should throw if the response does not match the schema", async () => {
    const scope = nock("https://api.apprentissage.beta.gouv.fr/api", {
      reqheaders: { authorization: "Bearer api-key" },
    })
      .get("/geographie/v1/mission-locale")
      .reply(200, { breaking: "schema" });

    const apiClient = new ApiClient({ key: "api-key" });
    const err = await apiClient.geographie
      .listMissionLocales()
      .then(() => {
        expect.unreachable("should throw an error");
      })
      .catch((error: ApiError) => {
        return error;
      });

    expect(err).toBeInstanceOf(ApiParseError);
    expect(err.name).toBe("ApiParseError");
    expect(err.message).toMatchSnapshot();

    expect(scope.isDone()).toBe(true);
  });

  it("should accepts future schema ehancements", async () => {
    const scope = nock("https://api.apprentissage.beta.gouv.fr/api", {
      reqheaders: { authorization: "Bearer api-key" },
    })
      .get("/geographie/v1/mission-locale")
      .reply(
        200,
        response.map((c) => ({
          ...c,
          new_field: "new_field",
        }))
      );

    const apiClient = new ApiClient({ key: "api-key" });

    const data = await apiClient.geographie.listMissionLocales();

    expectTypeOf(data).toEqualTypeOf<IMissionLocale[]>();

    expect(scope.isDone()).toBe(true);
    expect(data).toEqual(response);
  });
});
