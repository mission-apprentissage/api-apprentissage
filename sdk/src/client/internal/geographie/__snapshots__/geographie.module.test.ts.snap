// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`listDepartements > should throw if the response does not match the schema 1`] = `
"api-alternance-sdk: Error while parsing API response, this is either a bug in the SDK or a change in the API response. Please report this issue.
{"errors":["Invalid input"]}"
`;

exports[`listMissionLocale > should throw if the response does not match the schema 1`] = `
"api-alternance-sdk: Error while parsing API response, this is either a bug in the SDK or a change in the API response. Please report this issue.
{"errors":["Invalid input"]}"
`;

exports[`rechercheCommune > should throw if the response does not match the schema 1`] = `
"api-alternance-sdk: Error while parsing API response, this is either a bug in the SDK or a change in the API response. Please report this issue.
{"errors":["Invalid input"]}"
`;
